frontend/public/components/monitoring/targets.tsx

```diff
@@ -378,6 +430,28 @@ const List: React.FC<ListProps> = ({ loaded, loadError, targets }) => {
             {loadError}
           </Alert>
         )}
+        {serviceMonitorWatchError && showServiceMonitorError && (
+          <Alert
+            className="co-alert"
+            title={t('public~Error loading service monitor data')}
+            variant="danger"
+            actionClose={
+              <AlertActionCloseButton onClose={() => setShowServiceMonitorError(false)} />
+            }
+          >
+            {serviceMonitorWatchError}
+          </Alert>
```

Comment:
How about defining a `WatchErrorAlert` component instead of having a `getWatchError` function?

```suggestion
          <WatchErrorAlert
            errorMessage={serviceMonitorWatchError}
            onClose={hideServiceMonitorError}
            title={t('public~Error loading service monitor data')}
          />
```